src/Discussion/Discussion.php

```diff
@@ -461,6 +464,24 @@ public static function setStateUser(User $user)
     protected function setTitleAttribute($title)
     {
         $this->attributes['title'] = $title;
-        $this->slug = Str::slug($title);
+        $app = Application::getInstance();
+        $settings = $app->make(SettingsRepositoryInterface::class);
+        $locale = $settings->get('default_locale') ?? 'en';
+        $this->slug = Str::slug($title, '-', $locale);
+    }
+
+    /**
+     * Get the description.
+     *
+     * Returns the stored description (if any) or generates one based on startPost´s content.
+     *
+     * @param string $description
+     * @return string
+     */
+    protected function getDescriptionAttribute($description)
+    {
+        $description = ($description == null) ? $this->startPost->content : $description;
```

Comment:
This is gonna be inefficient when listing discussions D:

Reply:
@tobscure Are you referring to `$this->startPost->content`, right? The other option is to always have one (from discussion creation) on the database, not only when you (i.e.: via 3rd party extension) want to have a custom non-generated description there.

Reply:
Yeah I think we need to do something like this - any better ideas @franzliedke?

Reply:
The line we're talking about could also have been improved with the null coalescence operator.

I, however, agree to store a generated description, but only if none is set beforehand.